db/migrate/20211203153422_correct_datamodel_by_importing_neighbourhoods_json.rb

```diff
@@ -0,0 +1,73 @@
+class CorrectDatamodelByImportingNeighbourhoodsJson < ActiveRecord::Migration[6.1]
+  # This code is hecking UGLY
+  def create_unit(json_unit, parent_neighbourhood, parent_unit)
+    structure = { name: json_unit['properties']['name'],
+                  unit: json_unit['properties']['unit'],
+                  unit_code_key: json_unit['properties']['unit_code_key'],
+                  unit_code_value: json_unit['properties']['code'],
+                  unit_name: json_unit['properties']['name'] }
+
+    # We do not want to recreate wards that already exist, but we don't want to waste our
+    # cycles finding unit levels that don't exist
+    # Ergo, if the parent unit is a district that means that we are at ward level,
```

Comment:
And previously we only had word level data, so all other levels are just create?

Reply:
Yeah I explicitly remove all of the non-ward level units because the sourcing on it was "bad" -- i.e. we had Scotland listed as a region because the region name for the one ward in the original-original dataset was 'scotland'